src/pages/costModelsDetails/components/costModelsDetailsToolbar.tsx

```diff
@@ -0,0 +1,387 @@
+import {
+  InputGroup,
+  InputGroupText,
+  Pagination,
+  PaginationProps,
+  Select,
+  SelectOption,
+  SelectOptionProps,
+  SelectVariant,
+  TextInput,
+} from '@patternfly/react-core';
+import {
+  DataToolbar,
+  DataToolbarContent,
+  DataToolbarFilter,
+  DataToolbarGroup,
+  DataToolbarItem,
+} from '@patternfly/react-core/dist/esm/experimental';
+import { FilterIcon, SearchIcon } from '@patternfly/react-icons';
+import React from 'react';
+import { InjectedTranslateProps, translate } from 'react-i18next';
+import { Omit } from 'react-redux';
+import {
+  addMultiValueQuery,
+  addSingleValueQuery,
+  removeMultiValueQuery,
+  removeSingleValueQuery,
+} from '../../createCostModelWizard/filterLogic';
+
+interface SearchInputProps {
+  id: string;
+  value: string;
+  onChange: (value: string, event: React.FormEvent<HTMLInputElement>) => void;
+  onSearch: (evt: React.KeyboardEvent<HTMLInputElement>) => void;
+  placeholder?: string;
+}
+
+const SearchInput: React.SFC<SearchInputProps> = ({
+  id,
+  placeholder = '',
+  value,
+  onChange,
+  onSearch,
+}) => {
+  return (
+    <InputGroup>
+      <TextInput
+        value={value}
+        placeholder={placeholder}
+        id={id}
+        onChange={onChange}
+        onKeyPress={(evt: React.KeyboardEvent<HTMLInputElement>) => {
+          if (evt.key !== 'Enter' || value === '') {
+            return;
+          }
+          onSearch(evt);
+        }}
+      />
+      <InputGroupText style={{ borderLeft: '0' }}>
+        <SearchIcon />
+      </InputGroupText>
+    </InputGroup>
+  );
+};
+
+interface SelectFilterProps {
+  onSelect: (
+    event: React.MouseEvent<Element, MouseEvent> | React.ChangeEvent<Element>,
+    value: string
+  ) => void;
+  onToggle: (isExpanded: boolean) => void;
+  selected: string;
+  isExpanded: boolean;
+  options: SelectOptionProps[];
+}
+
+const SingleSelectFilter: React.SFC<SelectFilterProps> = ({
+  options,
+  onSelect,
+  onToggle,
+  selected,
+  isExpanded,
+}) => {
+  return (
+    <Select
+      isExpanded={isExpanded}
+      toggleIcon={<FilterIcon />}
+      variant={SelectVariant.single}
+      onSelect={onSelect}
+      onToggle={onToggle}
+      selections={selected}
+    >
+      {options.map(optionProps => (
+        <SelectOption key={`${optionProps.value}`} {...optionProps} />
+      ))}
+    </Select>
+  );
+};
+
+type PrimaryFilterBaseProps = InjectedTranslateProps &
+  Omit<SelectFilterProps, 'options'>;
+
+interface CostModelsDetailsToolberBaseProps extends InjectedTranslateProps {
+  primaryProps: Omit<PrimaryFilterBaseProps, 't'>;
+  paginationProps: PaginationProps;
+  secondaries: {
+    name: string;
+    deleteChip?: () => void;
+    render: () => React.ReactNode;
+  }[];
+  onClearAll: () => void;
+  onRemove: (category: string, chip: string) => void;
+  chips: { [k: string]: string | string[] };
+}
+
+const CostModelsDetailsToolberBase: React.SFC<CostModelsDetailsToolberBaseProps> = ({
+  t,
```

Comment:
Can this be replaced with `translate`?

Reply:
look at line 385-387:
https://github.com/project-koku/koku-ui/pull/1354/files/51e8b478c1eb6ab32210fd37a490a262205c1a08#diff-0e778d7b2fb39e96a09272e510669ff3R385-R387